runelite-client/src/main/java/net/runelite/client/plugins/questlist/QuestListPlugin.java

```diff
@@ -0,0 +1,401 @@
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package net.runelite.client.plugins.questlist;
+
+import com.google.inject.Provides;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.*;
+import net.runelite.api.events.*;
+import net.runelite.api.widgets.*;
+import net.runelite.client.callback.ClientThread;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.eventbus.Subscribe;
+import net.runelite.client.game.chatbox.ChatboxPanelManager;
+import net.runelite.client.game.chatbox.ChatboxTextInput;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.util.Text;
+
+import javax.inject.Inject;
+import java.util.*;
+
+@Slf4j
+@PluginDescriptor(
+	name = "Quest List",
+	description = "Show the location of the fairy ring teleport",
+	tags = {"quest"}
+)
+public class QuestListPlugin extends Plugin
+{
+	private static final int ENTRY_PADDING = 10;
+
+	private static final int SPRITE_SHOW_COMPLETED = SpriteID.OPTIONS_ZOOM_SLIDER_THUMB;
+	private static final int SPRITE_HIDE_COMPLETED = SpriteID.SQUARE_CHECK_BOX;
+
+	private static final String MENU_OPEN = "Open";
+	private static final String MENU_CLOSE = "Close";
+
+	private static final String MENU_SHOW = "Show";
+	private static final String MENU_HIDE = "Hide";
+
+	private static final String MENU_SEARCH = "Search";
+	private static final String MENU_COMPLETED = "Completed Quests";
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private QuestListConfig config;
+
+	@Inject
+	private ChatboxPanelManager chatboxPanelManager;
+
+	@Inject
+	private ClientThread clientThread;
+
+	private ChatboxTextInput searchInput = null;
+	private Widget questSearchButton;
+	private Widget questHideCompletedButton;
+
+	private EnumMap<QuestContainer, Collection<QuestWidget>> questSet;
+
+	private boolean hideCompleted = false;
```

Comment:
This should probably be persisted in a config

Reply:
I played around with this but found it undesirable for this setting to be persisted, especially when switching between accounts. So I chose to reset it upon login as that matched my expectations better. Would you still prefer for it to be persisted?

Reply:
why was it undesirable?

Reply:
On my main I have a lot of quests done so I hide the ones that are completed, resulting in a manageable list which is obviously filtered.

Then I switch to my alt, which has very few quests done (2 in total), so when looking at the list it's not immediately obvious some of them were hidden, leading me to scroll through the list once or twice looking for a specific quest before realizing the setting was persisted. I guess it could be persisted on a per-account basis?

But in general, when hiding information from the user, I prefer to only do so if they recently requested it. Otherwise it can cause confusion until they realize that they enabled a filter several days ago.

Reply:
Hmm. I don't want to add a per account setting for this until we have an actual framework for it, so I guess leaving it as-is is fine